insights/core/spec_factory.py

```diff
@@ -184,15 +184,18 @@ def validate(self):
         # 1. No Such File
         if not os.path.exists(self.path):
             raise ContentException("%s does not exist." % self.path)
-        # 2. No Filters for 'filterable=True' Specs
-        if (self.ds and filters.ENABLED and
-                any(s.filterable for s in dr.get_registry_points(self.ds)) and
-                not filters.get_filters(self.ds)):
-            raise NoFilterException("Skipping %s due to no filters." % dr.get_name(self.ds))
-        # 3. Customer Prohibits Collection
-        if not blacklist.allow_file("/" + self.relative_path):
-            log.warning("WARNING: Skipping file %s", "/" + self.relative_path)
-            raise BlacklistedSpec()
+        # 2. Check only when collecting
+        if isinstance(self.ctx, HostContext):
+            # 2.1 No Filters for 'filterable=True' Specs
+            if (self.ds and filters.ENABLED and
+                    isinstance(self.ctx, HostContext) and
```

Comment:
hey @xiangce , seems we can remove this `isinstance(self.ctx, HostContext)` checking , as it is assured by the `if` in line#188.

Reply:
Done. I forgot to remove it after level it up in the L188